pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -75,42 +69,37 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 }
 
 func ensureFirewallRule(ctx context.Context, gcpclient gcpclient.Interface, opt *Options) error {
-	firewall, err := getFirewallRule(ctx, gcpclient, opt)
+	firwallRule := firewallRule(opt, ingressAllowSSH)
+	firewall, err := getFirewallRule(ctx, gcpclient, opt, firwallRule.Name)
```

Comment:
1. function name should be plural - `ensureFirewallRules`
2. get the firewall rule name from the options, as mentioned in a different comment
3. why do you only check if one of the firewall rules exist? And if not, you create three different firewall rules. This might work for the happy path but is not very robust. Instead either get and then delete for all of them or ignore alreadyExists errors within the createFirewallRule function
4. I don't think there is a need to have the `firewallRule` function. Instead, have a slice with all the firewallRules. Then loop over this slice and get&create the firewall rule or create&ignoreNotFound errors

Reply:
refactor in 03a2330